allennlp/nlp_api.py

```diff
@@ -0,0 +1,289 @@
+from typing import Callable, Dict, List, Union
+
+import torch
+
+from .common.checks import ConfigurationError
+from .common import Params
+
+
+class NlpApi:
+    """
+    The ``NlpApi`` provides a high-level API for building NLP models.  You don't have to use it
+    for building a model with AllenNLP, but we think you should, because it's useful.
+
+    ``NlpApi`` abstracts away particular decisions that are frequently necessary in NLP, like how
```

Comment:
This is an attempt to illustrate the API - it's just the part that you use when you're building a model, not the part that you use when you're configuring an experiment.  That part will come later.

And the concrete objects are hiding in the constructor - all of the `embedder` and `recurrent_layer` dictionaries contain the concrete objects that get returned.